### PR TITLE
Add volumes and LB

### PR DESCRIPTION
**Pull request by jozuenoon**
Some initial work, refined to be use as wider internal service.

* added internal load balancer to route logs from outside cluster
* added persistent volume to store logs permamently in failure scenario

TODO:

* add template so it's possible to modify parameters from `values.yaml`

### CODE REVIEW

File: oklog/templates/oklog.yaml [hunk 1]
```diff
@@ -131,3 +156,13 @@ spec:
               containerPort: 7659
           resources:
 {{ toYaml .Values.oklog.resources | indent 12}}
+  volumeClaimTemplates:
```
Comment (jhaynie) [inline oklog/templates/oklog.yaml]: we probably need these to be configurable via Values.yaml

File: oklog/templates/oklog.yaml [hunk 2]
```diff
@@ -61,6 +61,31 @@ spec:
   selector:
     app: {{ template "pinpt.full_name" . }}-oklog
 
+---
```
Comment (jhaynie) [inline oklog/templates/oklog.yaml]: might want to make this opt-in/out in Values.yaml as well in case you don't need/want

**Review by jhaynie — COMMENTED** [review]
@jozuenoon thanks for your PR! appreciate it. 

@zachncst can you also review this?

LGTM with the exception of externalizing the config to values

File: oklog/templates/oklog.yaml (hunk 1 above)
Comment (zachncst) [reply oklog/templates/oklog.yaml]: Yea let's make this configurable. A good example is the postgres chart: https://github.com/kubernetes/charts/blob/master/stable/postgresql/templates/pvc.yaml

File: oklog/values.yaml [hunk 3]
```diff
@@ -22,6 +22,9 @@ oklog:
       memory: 128Mi
       cpu: 125m
 
+  serviceType: LoadBalancer  # [ClusterIP|LoadBalancer]
```
Comment (zachncst) [inline oklog/values.yaml]: This is duplicated from service.type. Maybe a bit confusing. If we need an internal-lb service that exposes different ports let's make it a subsection. Something like:

``` yaml
internalLBService: 
  enabled: false
  type: LoadBalancer
  ...
```
Defaults should be false.

File: oklog/templates/oklog.yaml (hunk 2 above)
Comment (zachncst) [reply oklog/templates/oklog.yaml]: Yea, I agree with this. Should put flags around it if the internal service is enabled.

**Review by zachncst — CHANGES_REQUESTED** [review]
Just some organizational changes requested.

**Comment by zachncst** [comment]
Sorry initial review comments may not have made sense, I didn't realize this was a public pull. Edited to be clearer.

**Comment by jozuenoon** [comment]
Sure will try to fix things as I get some spare time :)

**Comment by jozuenoon** [comment]
Done some changes. Opt in/out is now possible for LB and persistent data. Testing seems ok:
```bash
$ helm install --dry-run --debug oklog
```
(well with `yaml` no guarantees)

Other question is that maybe it would be nice to split this `oklog.yaml` into individual files and also split `ingest` and `store` into separate `StatefulSet`'s as I see now with moderate log capacity ~10GB per day that querying becomes quite hard... (getting `206 Partial Content` frequently).

File: oklog/templates/oklog.yaml [hunk 4]
```diff
@@ -113,8 +141,10 @@ spec:
             - /bin/sh
             - /go/bin/start.sh
           volumeMounts:
-            # - name: export
-            #   mountPath: /data
+{{- if and .Values.persistence.enabled }}
```
Comment (zachncst) [inline oklog/templates/oklog.yaml]: Think you left an and here.

File: oklog/templates/oklog.yaml [hunk 5]
```diff
@@ -131,3 +161,21 @@ spec:
               containerPort: 7659
           resources:
 {{ toYaml .Values.oklog.resources | indent 12}}
+{{- if and .Values.persistence.enabled }}
```
Comment (zachncst) [inline oklog/templates/oklog.yaml]: And here.

**Review by zachncst — CHANGES_REQUESTED** [review]

File: oklog/templates/oklog.yaml (hunk 5 above)
Comment (jozuenoon) [reply oklog/templates/oklog.yaml]: Thanks! Good spot.

**Comment by jozuenoon** [comment]
Hi. I think most things were fixed. Let me know what you think @zachncst . I would also recommend to build your own `oklog` docker from latest master as I found `v0.3.0` a little buggy.

**Comment by zachncst** [comment]
Thanks, @jozuenoon!

**Pull request merged**